documentation/modules/ROOT/pages/03-demo.adoc

```diff
@@ -7,28 +7,879 @@
 
 == Demonstration
 
-Include here content related to potential existing demos: blogs, articles, recorded videos, walkthrough guides, tutorials.
+Let's take a look at the following example.
+We'll be looking at a Cabs ride-sharing application, that mimics real-world solutions of similar kind.
+The application is written in a popular Java framework.
 
-[#demo-video]
-=== Watch a demonstration
+=== Initial application
 
-In this video you can see xpto:
+[IMPORTANT]
+====
+The following example should be considered as suboptimal, most likely a counterexample!
```

Comment:
I'd word it differently. That this is an example of a legacy / monolith app. 

That you refactor into more fancy stuff 

I think I get why want to explain here the legacy part, and do pseydo refactors, but I am 100% sure if this is really helping or needed

Reply:
Yes, maybe I can add it's an example of a legacy monolith application.

Also, I get your point it's complex (as every legacy), and potentially could throw away people from reading further.

How about I reverse the order of the demo page? First, I'll briefly introduce the app purpose, saying it's a legacy monolith, and immediately afterward I could just show the refactored sequence diagram and showcase the operation with Knative. At last, I could go back, and show the details of the refactoring. In that way, the complex parts became an optional part for a very keen on details audience.

WDYT?

Reply:
>First, I'll briefly introduce the app purpose, saying it's a legacy monolith, and immediately afterward I could just show the refactored sequence diagram and showcase the operation with Knative.

I think this is a great idea!